src/main/java/com/google/sps/servlets/EnterQueue.java

```diff
@@ -61,15 +73,59 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             Key classKey = KeyFactory.stringToKey(classCode);
             Entity classEntity = datastore.get(txn, classKey);
 
-            ArrayList<String> updatedQueue = (ArrayList) classEntity.getProperty("studentQueue");
+            // Get date in mm/dd/yyyy format
```

Comment:
The `doPost` method is very long and has multiple responsibilities. Could this be broken up into multiple methods to make the code clearer and to follow the [Single Responsibility Principal](https://en.wikipedia.org/wiki/Single-responsibility_principle)? For example you could have `updateVisits()`, `updateQueue()`, ...